gpt4all-backend/llamamodel.cpp

```diff
@@ -147,7 +179,11 @@ void LLamaModel::prompt(const std::string &prompt,
     params.prompt.insert(0, 1, ' ');
 
     // tokenize the prompt
-    auto embd_inp = ::llama_tokenize(d_ptr->ctx, params.prompt, false);
+    std::vector<llama_token> embd_inp(params.prompt.size() + 4);
+    int n = llama_tokenize(d_ptr->ctx, params.prompt.c_str(), embd_inp.data(), embd_inp.size(), d_ptr->empty);
+    assert(n >= 0);
+    embd_inp.resize(n);
+    d_ptr->empty = false;
```

Comment:
Since this never seems to get reset can't you just use a static variable declared in the method?

Reply:
The value would persist through successive separate instances of the class, though, so only the first ever created instance would add the BOS. PoC: https://godbolt.org/z/qTTex41Pv

Reply:
Yup, it's actually correct as is. You want the BOS to be readded on each new class instance